rest/views.py

```diff
@@ -0,0 +1,134 @@
+"""
+Controllers for REST app
+"""
+
+from __future__ import unicode_literals
+
+from rest_framework.generics import (
+    ListCreateAPIView,
+    RetrieveUpdateDestroyAPIView,
+)
+from rest_framework.permissions import (
+    DjangoObjectPermissions,
+    IsAuthenticated,
+)
+from rest_framework.reverse import reverse
+
+from rest.serializers import (
+    VocabularySerializer,
+    RepositorySerializer,
+    TermSerializer,
+)
+from learningresources.models import Repository
+
+
+# pylint: disable=too-many-ancestors
+class RepositoryList(ListCreateAPIView):
+    """REST list view for Repository"""
+    permission_classes = (DjangoObjectPermissions,)
+    queryset = Repository.objects.all()
+    serializer_class = RepositorySerializer
+    lookup_field = 'slug'
+    lookup_url_kwarg = 'repo_slug'
+
+    def get_success_headers(self, data):
+        """Add Location header for model create"""
+        url = reverse('repository-detail', args={'repo_slug': data['slug']})
+        return {'Location': url}
+
+
+class RepositoryDetail(RetrieveUpdateDestroyAPIView):
+    """REST detail view for Repository"""
+    permission_classes = (DjangoObjectPermissions,)
+    serializer_class = RepositorySerializer
+    lookup_field = 'slug'
+    lookup_url_kwarg = 'repo_slug'
+
+    def get_queryset(self):
+        """Filter to this repository"""
+        return Repository.objects.filter(slug=self.kwargs['repo_slug'])
+
+
+class VocabularyList(ListCreateAPIView):
+    """REST list view for Vocabulary"""
+    serializer_class = VocabularySerializer
+    lookup_field = 'slug'
+    lookup_url_kwarg = 'vocab_slug'
+    # WARNING: when implement authorization this should be changed
+    permission_classes = (IsAuthenticated,)
+
+    def get_queryset(self):
+        """Filter to vocabularies for a repository"""
+        return Repository.objects.get(
+            slug=self.kwargs['repo_slug']
+        ).vocabulary_set.all()
```

Comment:
There is extra work to be done to protect this: George and I talked and he is on it